### DISCUSSION TITLE
Integration with COZI Calendar - using emoji

### DISCUSSION THREAD

**Discussion by theRyanElliott** [eouia/MMM-CalendarExt2#302] [Q&A]
I have this calendar module setup and configured to work with my COZI family calendar.  It's working fine, except not displaying the emoji's I put in the calendar.  I'm using emoji's for youngest as he's still learning to read.

Is there anyway to get emoji's on COZI calendar to display?

**Comment by klaernie** [comment]
Does the font in use contain the emoji you're using? If not you should take a look if you can find a font that contains these glyphs, and set it using CSS (maybe just limited to that calendar).

How do these events look like in the ICS?

**Reply by theRyanElliott** [reply]
I did not change the default font value.  But I will look into that

**Reply by klaernie** [reply]
From looking at the support docs, I'd assume you put emojis in using your iOS/Android keyboard.
If the calendar events are a finite repeating set, you could also write a `transform` for the view showing this calendar, that assigns the right emoji based on the title text. Bonus point would then be the still-visible text, that your youngest can associate with the item and probably actually read that work way earlier ;)
But you could also use the same technique to replace a Unicode emoji in the text title into the `event.icon`, then stripping that character from the text. But that also is probable quite an effort if you're maintaining more than 10 kinds of events for your youngest.

**Reply by theRyanElliott** [reply]
I ended up going with a different calendar module, and also ended up including an emoji-supported font in the fonts directory, and using that font through the custom.css, for the entire calendar.  Now it shows exactly how I want, and my 6 year old has no problem telling what events are his over everyone elses.

**Reply by klaernie** [reply]
Sure, just out of curiosity, for which module did you go?

**Reply by theRyanElliott** [reply]
@klaernie I ended up using the MMM-CalendarMonthly module from kolbyjack, and adding the Segoe UI Emoji font.

**Comment by KristjanESPERANTO** [comment]
For anyone coming across this discussion: emojis should work out of the box as long as an emoji-capable font is installed on your system. The module renders event titles as plain HTML, so there's no special handling needed on our end.

If emojis are not showing, installing an emoji font usually does the trick:

```bash
# Debian/Raspberry Pi
sudo apt install fonts-noto-color-emoji
```

On Windows `Segoe UI Emoji` and on macOS `Apple Color Emoji` are typically already available.

:slightly_smiling_face:
